In [2]:
import numpy as numpy
import pandas as pd

# Examining the data frame 

In [3]:
perscore_df = pd.read_csv("personality_scores.csv" , delimiter = ";")
perscore_df.head(5)

ID Section 5 of 6 [I am always prepared.]  \
0   0                                 (3, 5)   
1   1                                 (3, 5)   
2   2                                 (3, 5)   
3   3                                 (3, 5)   
4   4                                 (3, 3)   

  Section 5 of 6 [I am easily disturbed.]  \
0                                  (4, 5)   
1                                  (4, 5)   
2                                  (4, 3)   
3                                  (4, 5)   
4                                  (4, 5)   

  Section 5 of 6 [I am exacting (demanding) in my work.]  \
0                                             (3, 5)       
1                                             (3, 5)       
2                                             (3, 3)       
3                                             (3, 3)       
4                                             (3, 3)       

  Section 5 of 6 [I am full of ideas.]  \
0                               (5, 5)   
1                               (5, 5)   
2                               (5, 5)   
3                               (5, 5)   
4                               (5, 3)   

  Section 5 of 6 [I am interested in people.]  \
0                                      (2, 3)   
1                                      (2, 5)   
2                                      (2, 5)   
3                                      (2, 5)   
4                                      (2, 3)   

  Section 5 of 6 [I am not interested in abstract ideas.]  \
0                                             (5, 3)        
1                                             (5, 3)        
2                                             (5, 5)        
3                                             (5, 3)        
4                                             (5, 3)        

  Section 5 of 6 [I am not interested in other people's problems.]  \
0                                             (2, 3)                 
1                                             (2, 5)                 
2                                             (2, 5)                 
3                                             (2, 3)                 
4                                             (2, 3)                 

  Section 5 of 6 [I am not really interested in others.]  \
0                                             (2, 5)       
1                                             (2, 5)       
2                                             (2, 5)       
3                                             (2, 3)       
4                                             (2, 3)       

  Section 5 of 6 [I am quick to understand things.]  ... Unnamed: 60  \
0                                            (5, 5)  ...         NaN   
1                                            (5, 5)  ...         NaN   
2                                            (5, 5)  ...         NaN   
3                                            (5, 3)  ...         NaN   
4                                            (5, 5)  ...         NaN   

  Unnamed: 61 Unnamed: 62 Unnamed: 63 Unnamed: 64 Unnamed: 65 Unnamed: 66  \
0         NaN         NaN         NaN         NaN         NaN         NaN   
1         NaN         NaN         NaN         NaN         NaN         NaN   
2         NaN         NaN         NaN         NaN         NaN         NaN   
3         NaN         NaN         NaN         NaN         NaN         NaN   
4         NaN         NaN         NaN         NaN         NaN         NaN   

  Unnamed: 67 Unnamed: 68 IPIP_HIGH_RISK  
0         NaN         NaN            NaN  
1         NaN         NaN            NaN  
2         NaN         NaN            NaN  
3         NaN         NaN            NaN  
4         NaN         NaN            NaN  

[5 rows x 70 columns]

# Checking Missing values

In [4]:
perscore_df.isnull().sum()
perscore_df.head(5)

ID Section 5 of 6 [I am always prepared.]  \
0   0                                 (3, 5)   
1   1                                 (3, 5)   
2   2                                 (3, 5)   
3   3                                 (3, 5)   
4   4                                 (3, 3)   

  Section 5 of 6 [I am easily disturbed.]  \
0                                  (4, 5)   
1                                  (4, 5)   
2                                  (4, 3)   
3                                  (4, 5)   
4                                  (4, 5)   

  Section 5 of 6 [I am exacting (demanding) in my work.]  \
0                                             (3, 5)       
1                                             (3, 5)       
2                                             (3, 3)       
3                                             (3, 3)       
4                                             (3, 3)       

  Section 5 of 6 [I am full of ideas.]  \
0                               (5, 5)   
1                               (5, 5)   
2                               (5, 5)   
3                               (5, 5)   
4                               (5, 3)   

  Section 5 of 6 [I am interested in people.]  \
0                                      (2, 3)   
1                                      (2, 5)   
2                                      (2, 5)   
3                                      (2, 5)   
4                                      (2, 3)   

  Section 5 of 6 [I am not interested in abstract ideas.]  \
0                                             (5, 3)        
1                                             (5, 3)        
2                                             (5, 5)        
3                                             (5, 3)        
4                                             (5, 3)        

  Section 5 of 6 [I am not interested in other people's problems.]  \
0                                             (2, 3)                 
1                                             (2, 5)                 
2                                             (2, 5)                 
3                                             (2, 3)                 
4                                             (2, 3)                 

  Section 5 of 6 [I am not really interested in others.]  \
0                                             (2, 5)       
1                                             (2, 5)       
2                                             (2, 5)       
3                                             (2, 3)       
4                                             (2, 3)       

  Section 5 of 6 [I am quick to understand things.]  ... Unnamed: 60  \
0                                            (5, 5)  ...         NaN   
1                                            (5, 5)  ...         NaN   
2                                            (5, 5)  ...         NaN   
3                                            (5, 3)  ...         NaN   
4                                            (5, 5)  ...         NaN   

  Unnamed: 61 Unnamed: 62 Unnamed: 63 Unnamed: 64 Unnamed: 65 Unnamed: 66  \
0         NaN         NaN         NaN         NaN         NaN         NaN   
1         NaN         NaN         NaN         NaN         NaN         NaN   
2         NaN         NaN         NaN         NaN         NaN         NaN   
3         NaN         NaN         NaN         NaN         NaN         NaN   
4         NaN         NaN         NaN         NaN         NaN         NaN   

  Unnamed: 67 Unnamed: 68 IPIP_HIGH_RISK  
0         NaN         NaN            NaN  
1         NaN         NaN            NaN  
2         NaN         NaN            NaN  
3         NaN         NaN            NaN  
4         NaN         NaN            NaN  

[5 rows x 70 columns]

# Checking for duplicates

In [24]:
#Checking if there are duplicates

perscore_df["ID"].is_unique

True

In [28]:
#Checking the number of duplicates
perscore_df["ID"].nunique()

1555

There are 1555 duplicates in the ID of the dataframe

# Dropping unnecessary columns

In [30]:
for score in perscore_df :
    if "Unnamed:" in score:
        del perscore_df[score]

In [31]:
perscore_df.head(5)

ID Section 5 of 6 [I am always prepared.]  \
0   0                                 (3, 5)   
1   1                                 (3, 5)   
2   2                                 (3, 5)   
3   3                                 (3, 5)   
4   4                                 (3, 3)   

  Section 5 of 6 [I am easily disturbed.]  \
0                                  (4, 5)   
1                                  (4, 5)   
2                                  (4, 3)   
3                                  (4, 5)   
4                                  (4, 5)   

  Section 5 of 6 [I am exacting (demanding) in my work.]  \
0                                             (3, 5)       
1                                             (3, 5)       
2                                             (3, 3)       
3                                             (3, 3)       
4                                             (3, 3)       

  Section 5 of 6 [I am full of ideas.]  \
0                               (5, 5)   
1                               (5, 5)   
2                               (5, 5)   
3                               (5, 5)   
4                               (5, 3)   

  Section 5 of 6 [I am interested in people.]  \
0                                      (2, 3)   
1                                      (2, 5)   
2                                      (2, 5)   
3                                      (2, 5)   
4                                      (2, 3)   

  Section 5 of 6 [I am not interested in abstract ideas.]  \
0                                             (5, 3)        
1                                             (5, 3)        
2                                             (5, 5)        
3                                             (5, 3)        
4                                             (5, 3)        

  Section 5 of 6 [I am not interested in other people's problems.]  \
0                                             (2, 3)                 
1                                             (2, 5)                 
2                                             (2, 5)                 
3                                             (2, 3)                 
4                                             (2, 3)                 

  Section 5 of 6 [I am not really interested in others.]  \
0                                             (2, 5)       
1                                             (2, 5)       
2                                             (2, 5)       
3                                             (2, 3)       
4                                             (2, 3)       

  Section 5 of 6 [I am quick to understand things.]  ...  \
0                                            (5, 5)  ...   
1                                            (5, 5)  ...   
2                                            (5, 5)  ...   
3                                            (5, 3)  ...   
4                                            (5, 5)  ...   

  Section 5 of 6 [I pay attention to details.]  \
0                                       (3, 5)   
1                                       (3, 1)   
2                                       (3, 5)   
3                                       (3, 5)   
4                                       (3, 5)   

  Section 5 of 6 [I seldom feel blue (down).]  \
0                                      (4, 3)   
1                                      (4, 1)   
2                                      (4, 1)   
3                                      (4, 1)   
4                                      (4, 5)   

  Section 5 of 6 [I spend time reflecting on things.]  \
0                                             (5, 5)    
1                                             (5, 5)    
2                                             (5, 3)    
3                                             (5, 5)    
4                                             (5, 5)    

  Section 5 of 6 [I start conversations.]  \
0                                  (1, 3

# Setting the ID column to be the index of the dataframe

In [32]:
pers_index = perscore_df.set_index("ID")
pers_index.head(5)

Section 5 of 6 [I am always prepared.]  \
ID                                          
0                                  (3, 5)   
1                                  (3, 5)   
2                                  (3, 5)   
3                                  (3, 5)   
4                                  (3, 3)   

   Section 5 of 6 [I am easily disturbed.]  \
ID                                           
0                                   (4, 5)   
1                                   (4, 5)   
2                                   (4, 3)   
3                                   (4, 5)   
4                                   (4, 5)   

   Section 5 of 6 [I am exacting (demanding) in my work.]  \
ID                                                          
0                                              (3, 5)       
1                                              (3, 5)       
2                                              (3, 3)       
3                                              (3, 3)       
4                                              (3, 3)       

   Section 5 of 6 [I am full of ideas.]  \
ID                                        
0                                (5, 5)   
1                                (5, 5)   
2                                (5, 5)   
3                                (5, 5)   
4                                (5, 3)   

   Section 5 of 6 [I am interested in people.]  \
ID                                               
0                                       (2, 3)   
1                                       (2, 5)   
2                                       (2, 5)   
3                                       (2, 5)   
4                                       (2, 3)   

   Section 5 of 6 [I am not interested in abstract ideas.]  \
ID                                                           
0                                              (5, 3)        
1                                              (5, 3)        
2                                              (5, 5)        
3                                              (5, 3)        
4                                              (5, 3)        

   Section 5 of 6 [I am not interested in other people's problems.]  \
ID                                                                    
0                                              (2, 3)                 
1                                              (2, 5)                 
2                                              (2, 5)                 
3                                              (2, 3)                 
4                                              (2, 3)                 

   Section 5 of 6 [I am not really interested in others.]  \
ID                                                          
0                                              (2, 5)       
1                                              (2, 5)       
2                                              (2, 5)       
3                                              (2, 3)       
4                                              (2, 3)       

   Section 5 of 6 [I am quick to understand things.]  \
ID                                                     
0                                             (5, 5)   
1                                             (5, 5)   
2                                             (5, 5)   
3                                             (5, 3)   
4                                             (5, 5)   

   Section 5 of 6 [I am quiet around strangers.]  ...  \
ID                                                ...   
0                                         (1, 3)  ...   
1                                         (1, 3)  ...   
2                                         (1, 1)  ...   
3                                         (1, 3)  ...   
4                                         (1, 1)  ...   

   Section 5 of 6 [I pay attention to details.]  \
ID                                                
0                          

# Asserting new data frame with original dataframe

In [33]:
assert len(perscore_df) == len(pers_index)